torchgeo/datasets/deepglobelandcover.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""DeepGlobe Land Cover Classification Challenge dataset."""
+
+import os
+from typing import Callable, Dict, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import (
+    check_integrity,
+    draw_semantic_segmentation_masks,
+    extract_archive,
+    rgb_to_mask,
+)
+
+
+class DeepGlobeLandCover(VisionDataset):
+    """DeepGlobe Land Cover Classification Challenge dataset.
+
+    The `DeepGlobe Land Cover Classification Challenge <https://competitions.codalab.org/competitions/18468/>` dataset 
+    offers high-resolution sub-meter satellite imagery focusing for the task of semantic segmentation to detect areas of 
+    urban, agriculture, rangeland, forest, water, barren, and unknown.
+    It contains 1,146 satellite images of size 2448 x 2448 pixels in total, 
+    split into training/validation/test sets, the original dataset can be downloaded from 
+    `Kaggle <https://www.kaggle.com/datasets/balraj98/deepglobe-land-cover-classification-dataset>`.
+    However, we only use the training dataset with 803 images since the original test and valid dataset 
+    are not accompanied by labels. The dataset that we use with a custom train/test split can be downloaded from 
+    `Kaggle <https://www.kaggle.com/datasets/geoap96/deepglobe2018-landcover-segmentation-traindataset>`
+    (created as a part of Computer Vision by Deep Learning (CS4245) course offered at TU Delft). 
+
+    Dataset format:
+
+    * images are RGB data
+    * masks are RGB image with with unique RGB values representing the class
+
+    Dataset classes:
+
+    0. Urban land
+    1. Agriculture land
+    2. Rangeland
+    3. Forest land
+    4. Water
+    5. Barren land
+    6. Unknown
+
+    File names for satellite images and the corresponding mask image are id_sat.jpg and id_mask.png, 
+    where id is an integer assigned to every image. 
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * DeepGlobe 2018: A Challenge to Parse the Earth Through Satellite Images
+
+    """  
+
+    filename = "data.zip"
+    data_root = "data"
+    md5 = "f32684b0b2bf6f8d604cd359a399c061"
+    splits = ["train", "test"]
+    classes = [
+        "Urban land",
+        "Agriculture land",
+        "Rangeland",
+        "Forest land",
+        "Water",
+        "Barren land",
+        "Unknown"
+    ]
+    colormap = [
+        (0, 255, 255),
+        (255, 255, 0),
+        (255, 0, 255),
+        (0, 255, 0),
+        (0, 0, 255),
+        (255, 255, 255),
+        (0, 0, 0)
+    ]
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        checksum: bool = False,
```

Comment:
Can this dataset be automatically downloaded?

Reply:
We need to be signed into Kaggle for this.

Reply:
Could add an optional dep on https://pypi.org/project/kaggle/ like we do radiant-mlhub.